yapapi/engine.py

```diff
@@ -100,11 +102,8 @@ def __init__(
         budget: Union[float, Decimal],
         strategy: BaseMarketStrategy,
         event_consumer: Callable[[events.Event], None],
-        subnet_tag: Optional[str] = None,
-        payment_driver: Optional[str] = None,
-        payment_network: Optional[str] = None,
         stream_output: bool = False,
-        app_key: Optional[str] = None,
+        api_config: ApiConfig,
```

Comment:
well, again, see my comment about the `subnet_tag` and `payment_x` above... the way I see it, they should be separate from the ApiConfig... 

maybe the payment parameters could be gathered together in a `PaymentConfig` dataclass but that's another matter altogether and possible a topic for a future change...